server/src/main/java/io/crate/action/sql/Sessions.java

```diff
@@ -172,4 +173,12 @@ public void cancel(KeyData keyData) {
     public Iterable<Session> getActive() {
         return sessions.values();
     }
+
+    public Iterable<Cursor> getCursors(User user) {
+        return () -> sessions.values().stream()
+            .filter(session ->
+                user.isSuperUser() || session.sessionSettings().sessionUser().equals(user))
```

Comment:
Is this enough? Shouldn't we check for cluster privileges, similar to what we do here: https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/expression/scalar/HasPrivilegeFunction.java#L234
Can we also have some test for this?